### PR TITLE
Don't override any pattern by its repetition in _longopt

### PR DESCRIPTION
**Pull request by przepompownia**
The option --recursive for `grep` command did not show up
because it was overriden by --directories.

### CODE REVIEW

File: bash_completion [hunk 1]
```diff
@@ -1878,7 +1878,9 @@ _longopt()
 
     if [[ "$cur" == -* ]]; then
         COMPREPLY=( $( compgen -W "$( LC_ALL=C $1 --help 2>&1 | \
-            command sed -ne 's/.*\(--[-A-Za-z0-9]\{1,\}=\{0,1\}\).*/\1/p' | sort -u )" \
+            command grep --basic-regexp --line-number --only-matching -- '--[-A-Za-z0-9]\{1,\}=\{0,1\}' | \
```
Comment (scop) [inline bash_completion]: These options to `grep` are not portable, see CONTRIBUTING.md

File: bash_completion [hunk 2]
```diff
@@ -1878,7 +1878,9 @@ _longopt()
...
+            mawk -F: '{ if (!options[$1]) options[$1] = $2 } END { for (line in options) printf("%s\n", options[line]);  }' | \
```
Comment (scop) [inline bash_completion]: `mawk` isn't portable either, plain awk would be acceptable but I hope these commands can be simplified overall.

Comment (przepompownia) [reply bash_completion]: Right. Clear bash is sufficient.

**Comment by scop** [comment]
Applied with a clarified commit message and a few followup commits, thanks!

**Pull request merged**